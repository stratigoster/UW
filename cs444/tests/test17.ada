PACKAGE foo IS
  type Week is (Sun,Mon,Tue,Wed,Thu,Fri,Sat);
  type WeekEnd is (Sat,Sun);
  myFaveDay : Week := Sun; -- overloading works!
  worstDay : Week := Wed;
  today : WeekEnd := Sun;
  badDay : BOOLEAN := today = Mon AND today /= myFaveDay OR ELSE today = worstDay;
--                    ^^^^^^^^^^^     ^^^^^^^^^^^^^^^^^^         ^^^^^^^^^^^^^^^^
-- 3 incompatible enumerator type instances: can't compare Week and WeekEnd
END foo;
