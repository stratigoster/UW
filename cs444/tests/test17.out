***ERROR***:7:29:	Not compatible enumerator type instances
***ERROR***:7:45:	Not compatible enumerator type instances
***ERROR***:7:72:	Not compatible enumerator type instances

------
Compilation error
------
